src/doc/reference.md

```diff
@@ -1047,11 +1047,8 @@ This is a list of behavior not considered *unsafe* in Rust terms, but that may
 be undesired.
 
 * Deadlocks
-* Reading data from private fields (`std::repr`)
```

Comment:
I think it's fine to make a private `mem::uninitialized` as long as a public consumer can't observe it. Making a bool = 3 is still invalid, though.